sdk/tables/azure-data-tables/azure/data/tables/_generated/aio/operations/_operations.py

```diff
@@ -733,6 +704,12 @@ async def update_entity(  # pylint: disable=inconsistent-return-statements
             409: ResourceExistsError,
             304: ResourceNotModifiedError,
         }
+        if match_condition == MatchConditions.IfNotModified:
+            error_map[412] = ResourceModifiedError
+        elif match_condition == MatchConditions.IfPresent:
+            error_map[412] = ResourceNotFoundError
```

Comment:
Can we test (or check the recordings) whether calling `update_entity` ever results in a 412 for `ResourceNotFound`?

Reply:
We already have the tests for `delete_entity()` and `update_entity()` operations in storage and cosmos, but the status code in response are `404`. 
e.g.
https://github.com/Azure/azure-sdk-for-python/blob/6adefdcac04e52ae3672d21ef4b2392b7d94c3ff/sdk/tables/azure-data-tables/tests/test_table_entity.py#L830-L839
[recording](https://github.com/Azure/azure-sdk-assets/blob/python/tables/azure-data-tables_e1a287b0d2/python/sdk/tables/azure-data-tables/tests/recordings/test_table_entity.pyTestTableEntitytest_update_entity_not_existing.json#L70)

Reply:
That test is not the right one - this is the correct test here:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/tables/azure-data-tables/tests/test_table_entity.py#L869

Recording:
https://github.com/Azure/azure-sdk-assets/blob/python/tables/azure-data-tables_e1a287b0d2/python/sdk/tables/azure-data-tables/tests/recordings/test_table_entity.pyTestTableEntitytest_update_entity_with_if_doesnt_match.json#L153-L173

Reply:
@YalinLi0312 - did you take a look at the test above?

Reply:
I didn't check the result when entity exists, we can get 412 in this case. 

Update: added the same test but with entity does not exist to the same test function.